common/log/throttle.go

```diff
@@ -18,35 +18,31 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package loggerimpl
+package log
 
 import (
 	"math/rand"
 	"sync/atomic"
 
-	"github.com/uber/cadence/common/dynamicconfig"
-	"github.com/uber/cadence/common/log"
 	"github.com/uber/cadence/common/log/tag"
 	"github.com/uber/cadence/common/quotas"
 )
 
 type throttledLogger struct {
 	rps     int32
 	limiter quotas.Limiter
-	log     log.Logger
+	log     Logger
 }
 
-var _ log.Logger = (*throttledLogger)(nil)
-
 const skipForThrottleLogger = 6
 
 // NewThrottledLogger returns an implementation of logger that throttles the
 // log messages being emitted. The underlying implementation uses a token bucket
 // ratelimiter and stops emitting logs once the bucket runs out of tokens
 //
 // Fatal/Panic logs are always emitted without any throttling
-func NewThrottledLogger(logger log.Logger, rps dynamicconfig.IntPropertyFn) log.Logger {
-	var log log.Logger
+func NewThrottledLogger(logger Logger, rps func() int) Logger {
```

Comment:
I don't think it's better to have a naked function than the "sematic" `dynamicconfig.IntPropertyFn`

Reply:
This introduces circular dependency - dynamic config use log.Logger, so log package is not allowed to have dependency on the dynamicconfig. Also, rps is always called without params, so the annotation of IntPropertyFn does not help.

Reply:
I disagree it does not help. It says it's a property from dynamic config and not just a random function that returns an int. 

I see the issue with the cyclic dependency. That's probably why it was split :) I think it's ok since we have the issue.

Reply:
kinda looks like the underlying issue is that we have multiple complex implementations in `dynamicconfig` itself, which are importing other stuff in what's otherwise a dead-end package.  might be worth making e.g. `dynamicconfig/fileconfig` and `dynamicconfig/collection` packages

Reply:
Makes sense. I will follow up with a PR that moves parts of dynamicconfig and can restore this use.

Reply:
Implemented the move in https://github.com/cadence-workflow/cadence/pull/6780